apphub/multi_task_learning/uncertainty_weighted_loss/uncertainty_loss_tf.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import tempfile
+
+import cv2
+import tensorflow as tf
+from tensorflow.keras import layers
+from tensorflow.python.keras.models import Model
+
+import fastestimator as fe
+from fastestimator.backend import reduce_loss
+from fastestimator.dataset.data import cub200
+from fastestimator.op.numpyop import Delete
+from fastestimator.op.numpyop.meta import Sometimes
+from fastestimator.op.numpyop.multivariate import HorizontalFlip, LongestMaxSize, PadIfNeeded, ReadMat, ShiftScaleRotate
+from fastestimator.op.numpyop.univariate import Normalize, ReadImage, Reshape
+from fastestimator.op.tensorop import TensorOp
+from fastestimator.op.tensorop.loss import CrossEntropy
+from fastestimator.op.tensorop.model import ModelOp, UpdateOp
+from fastestimator.schedule import cosine_decay
+from fastestimator.trace.adapt import LRScheduler
+from fastestimator.trace.io import BestModelSaver
+from fastestimator.trace.metric import Accuracy, Dice
+
+
+class ReduceLoss(TensorOp):
+    """TensorOp to average loss for a batch"""
+    def forward(self, data, state):
+        return reduce_loss(data)
+
+
+class UncertaintyWeightedLoss(layers.Layer):
```

Comment:
is this not generic enough to make into a re-usable layer?

Reply:
it is just one idea from one paper, i don't think the usage of it is general enough.